sdk/src/Core/Amazon.Runtime/_pcl/ClientContext.pcl.cs

```diff
@@ -38,9 +39,18 @@ public ClientContext(string appID)
             {
                 _clientID = _appSetting.GetValue(APP_ID_KEY, ApplicationSettingsMode.Local);
                 if (string.IsNullOrEmpty(_clientID))
-                {
-                    _clientID = Guid.NewGuid().ToString();
-                    _appSetting.SetValue(APP_ID_KEY, _clientID, ApplicationSettingsMode.Local);
+                { 
+                    string newClientID = Guid.NewGuid().ToString();
+                    try
+                    {
+                        _appSetting.SetValue(APP_ID_KEY, newClientID, ApplicationSettingsMode.Local);
+                    }
+                    catch(Exception e)
+                    {
+                        Logger.GetLogger(typeof(ClientContext)).Error(e, "Failed to save client id to local application setting");
+                        throw;
```

Comment:
Logging is good, but I think we should also throw the exception with a helpful message.  Instead of simply rethrowing, could we wrap in a new exception and make the original exception an inner exception?

Reply:
done!